src/components/QuizGame/index.jsx

```diff
@@ -0,0 +1,97 @@
+import React, { useState } from "react";
+import { fetchQuizQuestions } from "../../components/QuizGameApi";
+import QuestionCard from "../../components/QuizGameQuestions";
+import "./index.css";
+
+const DIFFICULTY = {
+  EASY: "easy",
+  MEDIUM: "medium",
+  HARD: "hard",
+};
+const TOTAL_QUESTIONS = 10;
+
+const QuizGame = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [questions, setQuestions] = useState([]);
+  const [currentQuestionNumber, setCurrentQuestionNumber] = useState(0);
+  const [userAnswers, setUserAnswers] = useState([]);
+  const [score, setScore] = useState(0);
+  const [isGameStart, setIsGameStart] = useState(false);
+
+  const startQuiz = async () => {
+    setIsLoading(true);
+    setIsGameStart(true);
+    const newQuestions = await fetchQuizQuestions(
+      TOTAL_QUESTIONS,
+      DIFFICULTY.EASY
+    );
+    setQuestions(newQuestions);
+    setScore(0);
+    setUserAnswers([]);
+    setCurrentQuestionNumber(0);
+    setIsLoading(false);
+  };
+  const checkAnswer = (e) => {
+    if (isGameStart) {
+      const userChosenAnswer = e.currentTarget.value;
+      const checkCorrectAnswer =
+        questions[currentQuestionNumber].correct_answer;
+      const isCorrect = checkCorrectAnswer === userChosenAnswer;
+      if (isCorrect) {
+        setScore((prev) => prev + 1);
+        const answerObject = {
+          question: questions[currentQuestionNumber].question,
+          userChosenAnswer,
+          isCorrect,
+          correctAnswer: questions[currentQuestionNumber].correct_answer,
+        };
+        setUserAnswers((prev) => [...prev, answerObject]);
+      }
+    }
+  };
+  const nextQuestion = () => {
+    const nextQNum = currentQuestionNumber + 1;
+    if (nextQNum >= TOTAL_QUESTIONS) {
+      setIsGameStart(true);
```

Comment:
shouldn't this be isGameOver? or Maybe you should set it's value to false?

Reply:
i am not using isGameOver anymore, its just isgamestart

Reply:
and it true to be able to start the game again

Reply:
please write that in a comment above that. it is not expected to a reader that the game starts again when you end it, I would think there's a victory screen or something but again it's best if explanations go in code not in pr comments

Reply:
I see, so isGameStart refers to the start button, I thought it checks if the game started already or not